### ISSUE TITLE
Smoothing the `Q(h)` relation

### ISSUE THREAD

**Issue by SouthEndMusic** [Deltares/Ribasim#1919]
In the grand scheme of boosting performance by making the whole RHS of the Ribasim ODE problem C1 smooth, doing this for `TabulatedRatingCurve` is one of the steps. Currently the $Q(h)$ relation is given by piecewise linear interpolation of the input data. To make this more smooth, we can use a cubic polynomial that rounds out the corners. As proposed by @Huite, this is easiest to understand by hydrologist by taking some fixed small interval centered around each data point where the linear interpolation is replaced by a polynomial (some care must be taken half this interval is bigger than the distance between points in the $x$ direction). 

For the nicest integration this could be incorporated into `DataInterpolations.jl`. For some inspiration see [here](https://southendmusic.github.io/CachedInterpolations.jl/stable/construction/construction_smoothed_linear_interpolation/), which uses an interval around the data points whose size is relative to the distance between points instead of absolute as proposed here.

**Comment by Huite** [comment]
Commenting this for completeness, this paper

> Model smoothing strategies to remove microscale discontinuities and spurious secondary optima in objective functions in hydrological calibration
[Dmitri Kavetski](https://agupubs.onlinelibrary.wiley.com/authored-by/Kavetski/Dmitri), [George Kuczera](https://agupubs.onlinelibrary.wiley.com/authored-by/Kuczera/George)
First published: 08 March 2007 https://doi.org/10.1029/2006WR005195

primarily suggest using logistic smoothing. This has the upside of being better than C2, but I'd say with a significant cost since our Qh tables contain quite a lot of rows.

Basically, you need a logistic function for each jump of dQ/dh. For integration, to keep it continous, you need to sum as many integrals (a softplus) as there are jumps (because the function is only 0 or 1 in the limit as x to ±infinity). For e.g. 20 points, that means 20 softplus evaluations.

![Image](https://github.com/user-attachments/assets/650d79c3-bbad-48a7-b139-34188f7945ae)
